File: binV6/seventh_problem.rb
require 'json'
require 'date'

staffs = JSON.parse(File.read('input_data/staff.json'), symbolize_names: true)

latest_feedbacks = staffs.map do |staff|
  feedbacks = staff.dig(:result, :feedback).max_by do |data|
    Date.parse(data[:date])
  end

  {
    email: staff[:email],
    full_name: staff[:full_name],
    latest_comment: feedbacks&.fetch(:comment, 'No feedback available yet.')
  }
end

print JSON.pretty_generate({ data: latest_feedbacks })

File: bins/binsV1/binV22/seventh_problem.js
const fs = require("fs");

const staffs = JSON.parse(fs.readFileSync("input_data/staff.json", "utf8"));

const latestFeedbacks = staffs.map((staff) => {
  const latestFeedback = staff?.result?.feedback?.toSorted(
    (a, b) => new Date(b.date) - new Date(a.date),
  )[0];

  return {
    email: staff.email,
    full_name: staff.full_name,
    latest_comment: latestFeedback?.comment ?? "No available feedback yet.",
  };
});

console.log(JSON.stringify({ latestFeedbacks }, null, 2));
